runtime/jcl/common/jdk_internal_foreign_abi_UpcallStubs.cpp

```diff
@@ -48,7 +48,7 @@ extern "C" {
  * to be allocated on the heap for thunk (especially on AIX only with 4KB virtual memory).
  */
 jboolean JNICALL
-Java_jdk_internal_foreign_abi_UpcallStubs_freeUpcallStub0(JNIEnv *env, jobject receiver, jlong address)
+Java_jdk_internal_foreign_abi_UpcallStubs_freeUpcallStub0(JNIEnv *env, jclass clazz, jlong address)
```

Comment:
The casts of `address` to pointers (on lines 61 and 63) should go through `intptr_t` first (a single cast shouldn't change **both** the width and signed-ness of a value at once).
```C
	void **functionPtr = (void **)(intptr_t)address;
	void *thunkAddr = *functionPtr;
#else /* defined(AIXPPC) || (defined(S390) && defined(zOS)) */
	void *thunkAddr = (void *)(intptr_t)address;
```

Reply:
I just updated the code there against the suggestion above but  I will need to verify them as it affects all supported platforms

Reply:
I verified the upcall related test suite with the change on all supported platforms which works good as expected.